apollo-runtime/src/main/java/com/apollographql/android/impl/RealApolloCall.java

```diff
@@ -132,17 +157,53 @@ private RealApolloCall(Operation operation, HttpUrl serverUrl, Call.Factory http
   }
 
   @Override @Nonnull public ApolloCall<T> clone() {
-    return new RealApolloCall<>(operation, serverUrl, httpCallFactory, httpCache, moshi, responseBodyConverter, cache,
-        cacheControl);
+    return new RealApolloCall<>(operation, serverUrl, httpCallFactory, httpCache, cacheControl, moshi,
+        responseFieldMapper, customTypeAdapters, cache);
+  }
+
+  private Response<T> cachedResponse() {
+    if (cacheControl == CacheControl.NETWORK_ONLY) {
+      return null;
+    }
+
+    T cachedData = cachedData();
+    if (cachedData != null) {
+      return new Response<>(operation, cachedData, null);
+    }
+
+    if (cacheControl == CacheControl.CACHE_ONLY) {
+      return new Response<>(operation, null, null);
```

Comment:
pls use constants that represent what null means, its difficult to see if this is correct or not when there are 2 unnamed params of null null.  Also why did ew not write the whole thing in kotlin? :-) too late now I guess